### PR TITLE
shell_i: Add a FinderSync-based implementation #2340

### PR DESCRIPTION
**Pull request by jturcotte**
Here what I have already working, I'll go through the checkmarks at https://github.com/owncloud/client/issues/2340#issuecomment-108480616 meanwhile.

cc: @guruz, @dragotin

### CODE REVIEW

**Comment by jturcotte** [comment]
Also please don't merge the request until we can try to build it without an official ownCloud signing key. I'll try at home but if you try to build it that should also do it.

**Comment by jturcotte** [comment]
Ok everything should be working fine, I think that the pull request is ready to be merged once you're OK with the code.

File: admin/osx/sign_app.sh [hunk 1]
```diff
@@ -1,12 +1,18 @@
 #!/bin/sh -xe
```
Comment (guruz) [inline admin/osx/sign_app.sh]: @danimo needs to comment on this :-)

File: shell_integration/MacOSX/OwnCloudFinder/RequestManager.m [hunk 2]
```diff
@@ -16,26 +16,32 @@
...
 
 		_registeredPathes = [[NSMutableDictionary alloc] init];
 		_requestedPaths = [[NSMutableSet alloc] init];
 
 		_shareMenuTitle = nil;
 
-		[NSTimer scheduledTimerWithTimeInterval:5 target:self selector:@selector(start) userInfo:nil repeats:YES];
+		// The NSConnection will block until the distant object came back and this creates a loop hanging Finder.
+		// Start from a timer to have time to unwind the stack first.
+		[NSTimer scheduledTimerWithTimeInterval:0 target:_syncClientProxy selector:@selector(start) userInfo:nil repeats:NO];
```
Comment (guruz) [inline shell_integration/MacOSX/OwnCloudFinder/RequestManager.m]: Btw, could also use ```performSelector``` with ```afterDelay``` (no need to change though)

http://woboq.com/blog/objective-c-for-qt-developers.html

**Comment by guruz** [comment]
Code wise i think it looks good :)

Will try it out now on my 10.9

**Comment by guruz** [comment]
It works nicely in 10.9. Great job @jturcotte 

@danimo 
As discussed this needs changes in the build server on signing:
https://github.com/owncloud/client/pull/3347/files#diff-af3b638bc2a3e6c650974192a53c7291R122
https://github.com/owncloud/client/pull/3347/files#diff-d451e3098afdc01086f319f531109038R3

We can merge when @danimo gives the OK for this..

**Comment by jturcotte** [comment]
We can also start by first enabling 2.0 builds without this branch.

Then I can fix rotor myself, merge the branch and make adjustments right away if anything isn't working as expected.

File: config.h.in [hunk 3]
```diff
@@ -5,9 +5,8 @@
 #cmakedefine WITH_QTKEYCHAIN 1
 #cmakedefine WITH_CRASHREPORTER
 #cmakedefine CRASHREPORTER_EXECUTABLE "@CRASHREPORTER_EXECUTABLE@"
+#define SOCKETAPI_TEAM_IDENTIFIER_PREFIX "@SOCKETAPI_TEAM_IDENTIFIER_PREFIX@"
 
-
-#cmakedefine GIT_SHA1 "@GIT_SHA1@"
```
Comment (danimo) [inline config.h.in]: Why remove this?

**Comment by danimo** [comment]
:+1:

**Pull request merged**